python-package/lightgbm/sklearn.py

```diff
@@ -412,6 +412,9 @@ def __init__(
         random_state: Optional[Union[int, np.random.RandomState]] = None,
         n_jobs: Optional[int] = None,
         importance_type: str = 'split',
+        use_early_stopping: bool = False,
```

Comment:
The Dask and `scikit-learn` interfaces should be identical.

Since `HistGradientBoosterClassifier` calls this `early_stopping` ([docs link](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.HistGradientBoostingClassifier.html)), please also call it `early_stopping` here and in the Dask interface.

Reply:
Also why is this defaulting to `False`? This PR's title says "enable early stopping automatically", but if it was merged as-is early stopping would be OFF by default.

Since the goal of this PR is to make LightGBM's scikit-learn interface consistent with scikit-learn itself, please follow what `HistGradientBoostingClassifier` does. As seen in their docs: https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.HistGradientBoostingClassifier.html

> **early_stopping**: `‘auto’` or `bool`, `default=’auto’`
.
> If ‘auto’, early stopping is enabled if the sample size is larger than 10000. If True, early stopping is enabled, otherwise early stopping is disabled.

Please match that behavior.

Reply:
This parameter has been removed from the public interface.